backend/src/prompts/templates/intent.j2

```diff
@@ -6,10 +6,21 @@ The question is:
 {{ question }} 
 
  
-The task is to comprehend the intention of the question. The question can be composed of different intents and when it is the case, examine all intents one by one to determine which one to tackle first as you may need the data gathered from a secondary intent to perform the first intent.  
-You are NOT ALLOWED to make up sample data or example values. Only use concrete data for which you can name the source. 
+Your task is to accurately comprehend the intentions behind the question. The question can be composed of different intents and when it is the case, examine all intents one by one to determine which one to tackle first as you may need the data gathered from a secondary intent to perform the first intent.  
+If the question contains multiple intents, break them down into individual tasks, and specify the order in which these tasks should be tackled. The order should ensure that each intent is addressed in a logical sequence, particularly if one intent depends on the data obtained from another.
+You are NOT ALLOWED to make up sample data or example values. Only use concrete data for which you can name the source.
 Based on this understanding, the following query must be formulated to extract the necessary data, which can then be used to address the question. 
 
+Guidelines:
+1. Determine each distinct intent in the question.
+2. Sequence the intents: Identify which intent should be tackled first and which should follow.
+3. For each intent, specify:
+    - The exact operation required (e.g., "literal search", "filter + aggregation", "data visualization").
+    - The category of the question (e.g., "data-driven", "data presentation", "general knowledge").
+    - Any specific parameters or conditions that apply.
+    - The correct aggregation and sorting methods if applicable.
+4. Avoid conflating intents: If a user’s query asks for data retrieval and its visualization, treat these as separate operations.

```

Comment:
The apostrophe at _user's query_ is flagging on my machine, probably because it's a single quote. Nit picking sorry but please can you use the same apostrophe as on line 1

Reply:
Done.